tests/test_main.py

```
@@ -917,6 +917,75 @@ def _connect_meshtastic(*_args, **_kwargs):
+    def test_main_shutdown_cancels_reconnect_before_ble_disconnect_and_unsubscribes(
+        self,
+    ):
+        """Shutdown should cancel reconnect before BLE disconnect and unsubscribe callbacks."""
+
+        import mmrelay.meshtastic_utils as mu
+
+        mock_iface = MagicMock()
+        reconnect_task = MagicMock()
+        mu.reconnect_task = reconnect_task
+
+        def _connect_meshtastic(*_args: Any, **_kwargs: Any) -> Any:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)